shim/shim.py

```diff
@@ -612,147 +612,148 @@ def _add_debug_arguments(parser: argparse.ArgumentParser) -> None:
+        options_json["field-models-paths"] = arguments.field_models_paths
 
     if arguments.literal_models_paths:
-        options.append("--literal-models-paths")
-        options.append(arguments.literal_models_paths)
+        options_json["literal-models-paths"] = arguments.literal_models_paths
 
     if arguments.proguard_configuration_paths:
-        options.append("--proguard-configuration-paths")
-        options.append(arguments.proguard_configuration_paths)
+        options_json["proguard-configuration-paths"] = arguments.proguard_configuration_paths
 
     if arguments.lifecycles_paths:
-        options.append("--lifecycles-paths")
-        options.append(arguments.lifecycles_paths)
+        options_json["lifecycles-paths"] = arguments.lifecycles_paths
 
     if arguments.shims_paths:
-        options.append("--shims-paths")
-        options.append(arguments.shims_paths)
+        options_json["shims-paths"] = arguments.shims_paths
 
     if arguments.graphql_metadata_paths:
-        options.append("--graphql-metadata-paths")
-        options.append(arguments.graphql_metadata_paths)
+        options_json["graphql-metadata-paths"] = arguments.graphql_metadata_paths
 
     if arguments.source_exclude_directories:
-        options.append("--source-exclude-directories")
-        options.append(arguments.source_exclude_directories)
+        options_json["source-exclude-directories"] = arguments.source_exclude_directories
 
     if arguments.generated_models_directory:
-        options.append("--generated-models-directory")
-        options.append(arguments.generated_models_directory)
+        options_json["generated-models-directory"] = arguments.generated_models_directory
 
     if arguments.sharded_models_directory:
-        options.append("--sharded-models-directory")
-        options.append(arguments.sharded_models_directory)
+        options_json["sharded-models-directory"] = arguments.sharded_models_directory
 
     if arguments.emit_all_via_cast_features:
-        options.append("--emit-all-via-cast-features")
+        options_json["emit-all-via-cast-features"] = arguments.emit_all_via_cast_features
+
     if arguments.propagate_across_arguments:
-        options.append("--propagate-across-arguments")
+        options_json["propagate-across-arguments"] = arguments.propagate_across_arguments
+
     if arguments.allow_via_cast_feature:
+        options_json["allow-via-cast-feature"] = []
         for feature in arguments.allow_via_cast_feature:
-            options.append("--allow-via-cast-feature=%s" % feature.strip())
+            options_json["allow-via-cast-feature"].append(feature.strip())
 
     if arguments.heuristics:
-        options.append("--heuristics")
-        options.append(arguments.heuristics)
+        options_json["heuristics"] = arguments.heuristics
 
     if arguments.sequential:
-        options.append("--sequential")
+        options_json["sequential"] = arguments.sequential
```

Comment:
nitpick here and a few other cases: if we entered that `if` then we know `arguments.sequential` is True, so let's just write:
```suggestion
        options_json["sequential"] = True
```